tinywav.c

```diff
@@ -166,23 +172,24 @@ void tinywav_close_read(TinyWav *tw) {
   tw->f = NULL;
 }
 
-size_t tinywav_write_f(TinyWav *tw, void *f, int len) {
+int tinywav_write_f(TinyWav *tw, void *f, int len) {
+  size_t samples_written = 0;
   switch (tw->sampFmt) {
     case TW_INT16: {
       int16_t *z = (int16_t *) alloca(tw->numChannels*len*sizeof(int16_t));
       switch (tw->chanFmt) {
         case TW_INTERLEAVED: {
           const float *const x = (const float *const) f;
           for (int i = 0; i < tw->numChannels*len; ++i) {
-            z[i] = (int16_t) (x[i] * 32767.0f);
+            z[i] = (int16_t) (x[i] * INT16_MAX);
```

Comment:
This should be `INT16_MAX-1`. And more pedantically, `(float) (INT16_MAX-1)`. Which is `32767.0f`. Why not leave it hardcoded as a number? Or are you worried about platforms where `INT16_MAX` is not 32768 😱?

Reply:
Wait a minute -- `INT16_MAX` **is** 32767 (!) , or in other words `=2^31-1` .. that's the whole point.

The fact that you got this wrong is exactly the reason why I think one should use these constants - that's why they exist. Semantically, we're "dividing the float sample value by the largest possible integer value" (thereby mapping +1.0 to this maximum value).

If somebody sees `32767`, are they going to think "oh, I'm super nerdy and I know this... this is the max value for 32 bit signed integers" .. what if they think it's 2^31 or the "max value minus 1" or something? Why take the risk of misunderstanding if the constant expresses the meaning of number more clearly?

... and about the `(float) INT16_MAX` .. yes, I guess it would make sense to add the explicit int-to-float promotion, since some compilers might issue a warning about this.

Reply:
... and about the `(float) INT16_MAX` .. yes, I guess it would make sense to add the explicit int-to-float promotion, since some compilers might issue a warning about this. 
On the other hand, since INT16_MAX is a `#define`, I can imagine the compiler can determine the number is not one of the very rare edge cases where an int cannot be stored in a float, and warn you while compiling IF it is a problem (e.g. the edge case)